teps/0119-add-taskrunspec-template-in-pipelinerun.md

```diff
@@ -0,0 +1,289 @@
+---
+status: proposed
+title: Add taskRunSpec template in PipelineRun
+creation-date: '2022-08-12'
+last-updated: '2022-08-12'
+authors:
+- '@yuzp1996'
+---
+
+# TEP-0119: Add taskRunSpec template in PipelineRun
+
+<!-- toc -->
+- [Summary](#summary)
+  - [Use Cases](#use-cases)
+- [Proposal](#proposal)
+- [Design Details](#design-details)
+  - [Former implement](#former-implement)
+  - [implement](#implement)
+  - [Test Plan](#test-plan)
+<!-- /toc -->
+
+## Summary
+User is supported to configure specify specific taskrun spec when they create pipelinerun. However, users is not supported to specify spec for all taskrun. 
+If you want to specify a common configuration for all taskrun they must specify them individually.
+
+This Tep is try to provide a convenient way that specify all taskrun spec in pipelinerun.
+
+### Use Cases
+Pipeline and Task User:
+I would like to use the same service account for all of my pipeline tasks except one.
+I want to specify compute resources that each TaskRun in my PipelineRun should run with, and don't want to have to specify them individually for each TaskRun.
+
+## Proposal
+Add filed taskRunSpecTemplate to PipelineRun.Spec then we can specify the common configuration in taskRunSpecTemplate, and the configuration will apply to all the TasRun.
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  generateName: clone-build-
+spec:
+  pipelineRef:
+    name: pipeline
+  taskRunSpecs:
+    - pipelineTaskName: clone
+      metadata:
+        annotations:
+          vault.hashicorp.com/agent-inject-secret-foo: "/path/to/foo"
+          vault.hashicorp.com/role: role-name
+// Add the following filed that same as taskRunSpecs
++ taskRunSpecTemplate:
++   metadata:
++     annotations:
++       vault.hashicorp.com/agent-inject-secret-foo: "/path/to/foo"
++       vault.hashicorp.com/role: role-name
++   taskServiceAccountName: xxx
++   stepOverrides: xxx
++   ....
+```
+## Design Details
+
+### previous implementation
+logic of TaskRunSpecTemplate will be like as TaskRunSpecs and the difference is that TaskRunSpecTemplate will apply to all taskrun while
+TaskRunSpecs will apply to one of the taskrun.
+
+Taskrun controller will be like that
+1. TaskRunSpecTemplate will apply to all taskrun
+2. TaskRunSpecs will apply to specify taskrun
+
+In this way the priority of taskrunSpecs will higher than TaskRunSpecTemplate
+
+
+Expand pipelinerun.Spec and add taskRunSpecTemplate
+
+
+```go
```

Comment:
no need to include all this code here-- that can be discussed in a pull request. Instead, I'd recommend including an example of a user-configured PipelineRun before this change, and the same PipelineRun after this change.

(A good example is in the matrix TEP: https://github.com/tektoncd/community/blob/main/teps/0090-matrix.md#substituting-string-parameters-in-the-tasks. There's an example of what a PipelineRun would look like with this feature, and the same PipelineRun without this feature.)